src/Query/Aggregation/Builder/CompositeAggregationQueryBuilder.php

```diff
@@ -0,0 +1,108 @@
+<?php
+declare(strict_types=1);
+
+namespace Kununu\Elasticsearch\Query\Aggregation\Builder;
+
+use Kununu\Elasticsearch\Exception\MissingAggregationAttributesException;
+use Kununu\Elasticsearch\Query\Aggregation\SourceProperty;
+use Kununu\Elasticsearch\Query\Aggregation\Sources;
+use Kununu\Elasticsearch\Query\CompositeAggregationQueryInterface;
+use Kununu\Elasticsearch\Query\Criteria\Filter;
+use Kununu\Elasticsearch\Query\Criteria\Filters;
+use Kununu\Elasticsearch\Query\QueryInterface;
+use Kununu\Elasticsearch\Query\RawQuery;
+use Kununu\Elasticsearch\Util\ArrayUtilities;
+
+final class CompositeAggregationQueryBuilder implements CompositeAggregationQueryInterface
+{
+    private ?array $afterKey;
+    private Filters $filters;
+    private ?string $name;
+    private ?Sources $sources;
+
+    private function __construct()
+    {
+        $this->afterKey = null;
+        $this->filters = new Filters();
+        $this->name = null;
+        $this->sources = null;
+    }
```

Comment:
What do you think about force directly the null instead of doing it on the constructor?

```suggestion
    private ?array $afterKey = null;
    private Filters $filters;
    private ?string $name = null;
    private ?Sources $sources = null;

    private function __construct()
    {
        $this->filters = new Filters();
    }
```